server/gamemodes/chess.py

```diff
@@ -76,22 +76,30 @@ def path_is_empty(
         victim = self.get_piece(rank, file)
         return victim.side != piece.side
 
+    def on_board(self, rank: int, file: int) -> bool:
+        """check if valid square i.e. rank and file on board"""
+        return (
+            rank >= 0 and rank <= 7
+            and file >= 0 and file <= 7
+        )
+
     def get_moves_in_direction(
             self, piece: models.Piece, rank_direction: int,
             file_direction: int) -> typing.Iterator[int, int]:
         """Get all moves for a unit in a direction."""
         rank, file = piece.rank, piece.file
         while True:
-            file += file_direction
             rank += rank_direction
+            file += file_direction
+            if not self.on_board(rank, file):
+                break
             target = self.get_piece(rank, file)
-            if target:
-                if target.side == piece.side:
-                    break
+            if (((not target) or (target.side != piece.side))
+                    and not self.hypothetical_check(piece.Side,
+                    (piece, rank, file))):
```

Comment:
```suggestion
            if (not target) or (target.side != piece.side):
```
`hypothetical_check` is already called by `possible_moves`.